parquet/src/main/java/org/apache/iceberg/parquet/ParquetVariantUtil.java

```diff
@@ -395,7 +399,30 @@ public Type object(VariantObject object, List<String> names, List<Type> typedVal
 
     @Override
     public Type array(VariantArray array, List<Type> elementResults) {
-      return null;
+      if (elementResults.isEmpty()) {
+        return null;
+      }
+
+      // Choose most common type as shredding type and build 3-level list
+      Type defaultTYpe = elementResults.get(0);
+      Type shredType =
+          elementResults.stream()
```

Comment:
This is okay, but it seems strange to me to rely on `equals` and `hashCode` for types and counting. It would be simpler to only shred values if there is a uniform type, which is when we would benefit from shredding.